src/3rd_party_add.c

```diff
@@ -82,7 +97,16 @@ enum swupd_code third_party_add_main(int argc, char **argv)
 		goto finish;
 	}
 
-	/* TODO implement */
+	/* The last two in reverse are the repo-name, repo-url */
+	if (add_repo_config(argv[argc - 2], argv[argc - 1])) {
+		ret = SWUPD_NO;
```

Comment:
I don't think SWUPD_NO is the appropriate code to return here, actually SWUPD_NO is considered a successful return code (currently only used in check-update).